src/rewrite.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { RewriteRule, RewriteRules } from "./rewrite";
+
+describe("RewriteRule", () => {
+  describe("without capture groups", () => {
+    it("applies the expected changes", () => {
+      const rule = new RewriteRule(/-[a-z0-9]+-?/gi, "");
+      expect(rule.apply("")).toEqual("");
+      expect(rule.apply("chicken")).toEqual("chicken");
+      expect(rule.apply("one-TWO-thRee-fOUr")).toEqual("onethRee");
```

Comment:
**Question** I feel I am missing something here. Looking at the regex, should this not be replaced to:
`one-TWORee-OUr`? 

The regex requires:
- One  -
- 1 or 2 of `[a-z0-9]`
- 0 or 1 -

Reply:
The `g` modifier in JavaScript regex syntax is global replace (perform as many replacements as possible), and the `i` modifier is case-insensitive replacement. The `g` is what results in the expected value.

Reply:
Is the `g` running it recursively as well?

Reply:
`g` isn't recursive. The search regex is `/-[a-z0-9]+-?/gi` and the replacement string is the empty string.

The input string is `one-TWO-thRee-fOUr`.
* `-TWO-` matches the regex. This gets replaced with the empty string, resulting in the input string being `onethRee-fOUr`.
* `-fOUr` matches the regex. This gets replaced with the empty string, resulting in the input string being `onethRee`.

Reply:
Ah, it's me not thinking clearly, `+` is 1 to many. All good!